definitions/procedures/pulp/remove.rb

```diff
@@ -23,16 +23,22 @@ def pulp_data_dirs
     end
 
     def pulp_packages
-      [
-        'pulp-server', 'python-pulp-streamer', 'pulp-puppet-plugins',
-        'python-pulp-rpm-common', 'python-pulp-common',
-        'pulp-selinux', 'python-pulp-oid_validation',
-        'python-pulp-puppet-common', 'python-pulp-repoauth',
-        'pulp-rpm-plugins', 'python-blinker', 'python-celery',
-        'python-django', 'python-isodate', 'python-ldap',
-        'python-mongoengine', 'python-nectar', 'python-oauth2',
-        'python-pymongo'
-      ]
+      all_possible = %w(pulp-admin-client pulp-agent pulp-consumer-client pulp-deb-admin-extensions
+                        pulp-deb-plugins pulp-docker-admin-extensions pulp-docker-plugins
+                        pulp-nodes-admin-extensions pulp-nodes-child pulp-nodes-common pulp-nodes-consumer-extensions
+                        pulp-nodes-parent pulp-ostree-admin-extensions pulp-ostree-plugins pulp-puppet-admin-extensions
+                        pulp-puppet-consumer-extensions pulp-puppet-handlers pulp-puppet-plugins pulp-puppet-tools
+                        pulp-python-admin-extensions pulp-python-plugins pulp-rpm-admin-extensions pulp-rpm-consumer-extensions
+                        pulp-rpm-handlers pulp-rpm-plugins pulp-rpm-yumplugins pulp-selinux pulp-server python-bson python-crane
+                        python-isodate python-mongoengine python-nectar python-pulp-agent-lib python-pulp-bindings python-pulp-client-lib
+                        python-pulp-common python-pulp-deb-common python-pulp-devel python-pulp-docker-common python-pulp-integrity
+                        python-pulp-manifest python-pulp-oid_validation python-pulp-ostree-common python-pulp-puppet-common
+                        python-pulp-python-common python-pulp-repoauth python-pulp-rpm-common python-pulp-streamer python-pymongo
+                        python-pymongo-gridfs python2-amqp python2-billiard python2-celery python2-debpkgr python2-django python2-kombu
+                        python2-solv python2-vine)
+
+      @installed_pulp_packages ||= all_possible.select { |pkg| find_package(pkg) }
+      @installed_pulp_packages
```

Comment:
here we only consider packages actually installed, and cache to not have to recalculate again when this method is called a 2nd time.